Automattic/wp-calypso
# ImportSignupFlow: Add confirm-import step

> Note: DO NOT MERGE - we need to make sure that, once in Calypso-proper, folk don't see the site preview there as well as in signup - this is being worked on currently.
> Note: The first 4 commits on this branch have been cherry-picked from #31588 (✅),  #31705 (✅) and #31643 (❌). Those 3 PRs should be merged first to help minimise the size of this PR, making for easier reviewing.

### Summary

This PR adds an extra step to the import signup flow which shows a preview of the targeted site and asks the customer to either confirm their ownership of the site, or, if it's not their site, go back to the previous step to try a different URL.

<img width="1045" alt="Screenshot 2019-03-25 at 12 10 31" src="https://user-images.githubusercontent.com/4335450/54918746-02a88d00-4ef7-11e9-9029-00e645f86df9.png">


### Testing Instructions

First, let's make sure that there are no regressions in our validation:

1. Apply this branch, then visit `/start/import/from-url`
1. Try submitting a random string  that is not a valid url (example: `not-a-valid-url`)
1. You should see an error notice, stating: "Please enter a valid URL."
1. Try submitting a url that is a valid site, but isn't importable (example: `google.com`)
1. You should see an error notice, stating: "We're not able to reach that site at this time, or it's not compatible with our URL importer."

Then, let's test the new step:

1. Try submitting a url that is a valid wix site (example: `https://smt593.wixsite.com/wowz`)
1. You should see that the button shows a busy state and a notice is shown telling you that we're checking to see if it's importable
1. Since the site is importable, we should then be taken directly to our new `confirm-import` step, where we should see the screen shown in the above screenshot.
1. Hit the negative button
1. You should be taken back to the url input step and it should function as before
1. Submit the same URL again, but this time hit the positive button to carry on with the signup flow
1. If signed out (on an incognito browser) you should be taken to the username & password step.
1. Follow this through until you land in Calypso. You'll notice that we're on the Wix import screen there, ready to confirm our import (future work will change this, so that we don't need to confirm the site preview twice)
